pallets/manta-pay/src/lib.rs

```diff
@@ -201,6 +201,9 @@ pub mod pallet {
         #[transactional]
         pub fn to_private(origin: OriginFor<T>, post: TransferPost) -> DispatchResultWithPostInfo {
             let origin = ensure_signed(origin)?;
+            for source in post.sources.iter() {
+                ensure!(asset_value_decode(*source) > 0u128, Error::<T>::BalanceLow);
```

Comment:
Also, are you sure that *each* source&sink must be nonzero?
Isn't it suffient for *at-least-one* source to be?

Reply:
we can set it to exist deposit of each token.

Reply:
not sure if existential deposit makes sense here.

Our private ledger doesn't have notion of ED so I can only see it on the `to_public` call.
But a sink for `to_public` must already exist, otherwise you can't pay the gas fees for the `to_public`, so you *can always* `to_public` amounts smaller than ED without risk of losing it.

Am I missing something?

Reply:
> Also, are you sure that _each_ source&sink must be nonzero? Isn't it suffient for _at-least-one_ source to be?

Ok, you're saying that maybe we don't loop all sources? In that case, the sources len will be only 1. I added another check that we discuss before.